test/components/pagination.js

```diff
@@ -0,0 +1,123 @@
+import { expect } from 'chai';
+import utils from '../utils';
+
+import Pagination from '../../lib/components/Pagination';
+
+describe('Component: Pagination', function () {
+
+	describe('Feature: Basics', function () {
+
+		it('renders correct number of pages', function () {
+			var props = {
+					currentPage: 1,
+					pageSize: 10,
+					total: 30
+				},
+				list = utils.getDOMNodeWithClass(Pagination, props, 'Pagination__list');
+
+			expect(list.children).to.have.length(3);
+		});
+
+		it('adds appropriate class to current page element', function () {
+			var props = {
+					currentPage: 2,
+					pageSize: 10,
+					total: 30
+				},
+				list = utils.getDOMNodeWithClass(Pagination, props, 'Pagination__list'),
+				selectedClass = 'is-selected',
+				classFilter = function (el) {
+					return el.getAttribute('class').includes(selectedClass);
+				};
+
+			expect(Array.from(list.children).filter(classFilter)[0].innerHTML).to.equal('2');
+		});
+
+		it('renders all pages', function () {
+			var props = {
+					currentPage: 1,
+					pageSize: 10,
+					total: 45
+				},
+				list = utils.getDOMNodeWithClass(Pagination, props, 'Pagination__list');
+
+			expect(Array.from(list.children, utils.getInnerHTML)).to.eql(['1', '2', '3', '4', '5']);
+		});
+
+	});
+
+	describe('Feature: Limit', function () {
+
+		let getPages = function (props) {
```

Comment:
I'll write a PR with legit tests;) The only thing I can say is that it does
support shallow rendering and the big thing for me is it makes your tests
readable. You can easily figure out what is being tested, I feel like that
does not create a higher barrier of entry, but a lower one.
On Mon, Sep 21, 2015 at 07:21 Nik Butenko <notifications@github.com> wrote:

> In test/components/pagination.js
> <https://github.com/elementalui/elemental/pull/58#discussion_r39975491>:
>
> > +		it('renders all pages', function () {
> > +			var props = {
> > +					currentPage: 1,
> > +					pageSize: 10,
> > +					total: 45
> > +				},
> > +				list = utils.getDOMNodeWithClass(Pagination, props, 'Pagination__list');
> > +
> > +			expect(Array.from(list.children, utils.getInnerHTML)).to.eql(['1', '2', '3', '4', '5']);
> > +		});
> > +
> > +	});
> > +
> > +	describe('Feature: Limit', function () {
> > +
> > +		let getPages = function (props) {
>
> const getPages = props => Array.from(utils.getDOMNodeWithClass(Pagination, props, 'Pagination__list').children, utils.getInnerHTML)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elementalui/elemental/pull/58/files#r39975491>.
>